### PR TITLE
PIN-3142 Make email mandatory when submitting an agreement

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala [hunk 1]
```diff
@@ -627,6 +628,15 @@ final case class AgreementApiServiceImpl(
           stamps = stamps.toManagement
         )
 
+    def validateConsumerEmails(agreement: PersistentAgreement): Future[Unit] =
+      for {
+        consumer <- tenantManagementService
+          .getTenantById(agreement.consumerId)
+        _        <- Future
+          .failed(ConsumerWithNotValidEmail(agreement.id, agreement.consumerId))
+          .whenA(consumer.mails.filter(_.kind == PersistentTenantMailKind.ContactEmail).isEmpty)
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]: don't use cats with future
```suggestion
        _        <- if (consumer.mails.filter(_.kind == PersistentTenantMailKind.ContactEmail).isEmpty) 
        Future.failed(ConsumerWithNotValidEmail(agreement.id, agreement.consumerId))
        else Future.unit
```

**Review by TonioGela — CHANGES_REQUESTED** [review]

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala [hunk 2]
```diff
@@ -643,6 +654,7 @@ final case class AgreementApiServiceImpl(
 
     for {
       uid    <- getUidFutureUUID(contexts)
+      _      <- if (agreement.state == Draft && newState == Pending) validateConsumerEmails(agreement) else Future.unit
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]: Agreement submission can also directly pass from Draft to Active
```suggestion
      _      <- if (agreement.state == Draft) validateConsumerEmails(agreement) else Future.unit
```

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala [hunk 3]
```diff
@@ -624,6 +625,16 @@ final case class AgreementApiServiceImpl(
           stamps = stamps.toManagement
         )
 
+    def validateConsumerEmails(agreement: PersistentAgreement): Future[Unit] =
+      for {
+        consumer <- tenantManagementService
+          .getTenantById(agreement.consumerId)
+        _        <-
+          if (consumer.mails.filter(_.kind == PersistentTenantMailKind.ContactEmail).isEmpty)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]:
```suggestion
          if (!consumer.mails.exists(_.kind == PersistentTenantMailKind.ContactEmail))
```

File: src/main/scala/it/pagopa/interop/agreementprocess/error/AgreementProcessErrors.scala [hunk 4]
```diff
@@ -99,5 +99,11 @@ object AgreementProcessErrors {
...
       extends ComponentError("0022", s"Attribute ${attributeId.toString} not found")
 
-  case object InvalidAttributeStructure extends ComponentError("0023", "Invalid Attribute Structure")
+  final case class ConsumerWithNotValidEmail(agreementId: UUID, tenantId: UUID)
+      extends ComponentError(
+        "0023",
+        s"Agreement ${agreementId.toString} has a consumer tenant ${tenantId.toString} with no valid email"
+      )
+
+  case object InvalidAttributeStructure extends ComponentError("0024", "Invalid Attribute Structure")
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/agreementprocess/error/AgreementProcessErrors.scala]: we should avoid to change error codes if not strictly required
```suggestion
  case object InvalidAttributeStructure extends ComponentError("0023", "Invalid Attribute Structure")
```

File: src/main/scala/it/pagopa/interop/agreementprocess/error/AgreementProcessErrors.scala [hunk 5]
```diff
@@ -99,5 +99,11 @@ object AgreementProcessErrors {
   final case class AttributeNotFound(attributeId: UUID)
       extends ComponentError("0022", s"Attribute ${attributeId.toString} not found")
 
-  case object InvalidAttributeStructure extends ComponentError("0023", "Invalid Attribute Structure")
+  final case class ConsumerWithNotValidEmail(agreementId: UUID, tenantId: UUID)
+      extends ComponentError(
+        "0023",
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/agreementprocess/error/AgreementProcessErrors.scala]:
```suggestion
        "0024",
```

**Review by galales — CHANGES_REQUESTED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**